bin/build_cmd.ml

```diff
@@ -231,6 +231,20 @@ let fmt =
     in
     let common, config = Common.init builder in
     let request (setup : Import.Main.build_system) =
+      let open Action_builder.O in
+      let* () =
+        if Lazy.force Lock_dev_tool.is_enabled
+        then
+          (* Note that generating the ocamlformat lockdir here means
+             that it will be created when a user runs `dune fmt` but not
+             when a user runs `dune build @fmt`. It's important that
+             this logic remain outside of `dune build`, as `dune
+             build` is intended to only build targets, and generating
+             a lockdir is not building a target. *)
```

Comment:
Is this text something that shows up in the actual code when doing this, or is it an explanation for the purposes of this documentation? Asking to see if I can format it to match the style guide or if that's the way it appears for the user.

Reply:
I think, this is the way at the moment. It could be rephrased and used for the documentation, when the feature is no more experimental and that implies updating the document for `dune build @fmt` and `dune fmt`.

cc: @gridbugs.